apis/operator/v1alpha1/discoveryservice_types.go

```diff
@@ -48,6 +49,11 @@ type DiscoveryServiceSpec struct {
 	// use since secret data is never shown in the logs.
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
 	Debug bool `json:"debug,omitempty"`
+	// Resources holds the Resource Requirements to use for the discovery service
+	// Deployment. When not set it defaults to no resource requests nor limits
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +optional
+	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
This is a pointer, following API conventions about optional parameters: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required.